pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/PravegaSensorCollectorIntegrationTests.java

```diff
@@ -60,9 +69,91 @@ public void tearDown() {
 
     }
 
+    @Test
+    @Timeout(value = 3, unit = TimeUnit.MINUTES)
+    @Order(1)
+    public void testPSCDataIntegration() {
+        try {
+            copyHelloWorldFile();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        final DeviceDriverManager deviceDriverManager = new DeviceDriverManager(properties);
+        Service startService = deviceDriverManager.startAsync();
+        try {
+            startService.awaitRunning(Duration.ofSeconds(30));
+            Thread.sleep(25000);
+        } catch (InterruptedException | TimeoutException e) {
+            throw new RuntimeException(e);
+        }
+        final Connection connection = SQliteDBUtility.createDatabase(properties.get("PRAVEGA_SENSOR_COLLECTOR_RAW1_DATABASE_FILE"));
+        final TransactionStateDB state = new TransactionStateSQLiteImpl(connection, null);
+
+        try {
+            List<FileNameWithOffset> completedFiles = state.getCompletedFileRecords();
+            Assertions.assertEquals(1, completedFiles.size());
+
+            URI controllerURI = setupUtils.getControllerUri();
+            String scope = properties.get("PRAVEGA_SENSOR_COLLECTOR_RAW1_SCOPE");
+            String streamName = properties.get("PRAVEGA_SENSOR_COLLECTOR_RAW1_STREAM");
+
+            StreamManager streamManager = StreamManager.create(controllerURI);
+
+            final String readerGroup = UUID.randomUUID().toString().replace("-", "");
+            final ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder()
+                    .stream(Stream.of(scope, streamName))
+                    .build();
+            try (ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scope, controllerURI)) {
+                readerGroupManager.createReaderGroup(readerGroup, readerGroupConfig);
+            }
+
+            try (EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(scope,
```

Comment:
Please refactor by extracting validation code in its own method. Also makes it reusable.